### PR TITLE
style(naming): replace word action with edit to clarify API, closes #50

### LINKED ISSUE #50: Using Action instead of Edit to more precisely refer to API
In this repo, we frequently use `actions` and `edits`.

However this repo only uses our newer core API and these should all be called `edits`.

Examples:

```tExtRef/unsubscribe.ts
29: * @returns An array of update and/or remove action representing changes required

tExtRef/subscribe.spec.ts
19:      const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
20:      expect(actions.length).to.equal(0);
28:      const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
29:      expect(actions.length).to.equal(0);
37:      const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
38:      expect(actions.length).to.equal(0);
45:      const actions = subscribe([{ sink, source: { fcda } }]);
46:      expect(actions.length).to.equal(0);
64:      const actions = subscribe([{ sink, source: { fcda, controlBlock } }], {
67:      expect(actions.length).to.equal(1);
72:    it("return update action for later binding type ExtRef ", () => {
77:      const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
78:      expect(actions.length).to.equal(1);
79:      expect(actions[0]).to.satisfies(isUpdate);
80:      expect((actions[0] as Update).element).to.equal(sink);
81:      expect((actions[0] as Update).attributes).to.deep.equal({
92:    it("returns insert action for non later binding type subscription", () => {
97:      const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
98:      expect(actions.length).to.equal(1);
99:      expect(actions[0]).to.satisfies(isInsert);
100:      expect(((actions[0] as Insert).parent as Element).tagName).to.equal(
103:      expect(((actions[0] as Insert).node as Element).tagName).to.equal(
106:      expect(((actions[0] as Insert).reference as Element).tagName).to.equal(
110:      const extRef = (actions[0] as Insert).node as Element;
172:      it("returns no update action with failing SPS/BOOLEAN check", () => {
177:        const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
178:        expect(actions.length).to.equal(0);
181:      it("returns no update action with failing CMV/FLOAT32 check ", () => {
186:        const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
187:        expect(actions.length).to.equal(0);
190:      it("returns update action with passing SPC/BOOLEAN check", () => {
195:        const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
196:        expect(actions.length).to.equal(1);
199:      it("returns update action with passing CMV/FLOAT32 check", () => {
204:        const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
205:        expect(actions.length).to.equal(1);
208:      it("return update action with srcxxx attributes with controlBlock set", () => {
215:        const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
216:        expect(actions.length).to.equal(1);
217:        expect(actions[0]).to.satisfies(isUpdate);
218:        expect((actions[0] as Update).element).to.equal(sink);
219:        expect((actions[0] as Update).attributes).to.deep.equal({
236:      it("return update action without srcXXX attributes with controlBlock set", () => {
243:        const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
244:        expect(actions.length).to.equal(1);
245:        expect(actions[0]).to.satisfies(isUpdate);
246:        expect((actions[0] as Update).element).to.equal(sink);
247:        expect((actions[0] as Update).attributes).to.deep.equal({
266:      it("return insert action to add ExtRef", () => {
273:        const actions = subscribe([{ sink, source: { fcda, controlBlock } }]);
274:        expect(actions.length).to.equal(2);
275:        expect(actions[0]).to.satisfies(isInsert);
276:        expect(((actions[0] as Insert).parent as Element).tagName).to.equal(
279:        expect(((actions[0] as Insert).node as Element).tagName).to.equal(
282:        expect((actions[0] as Insert).reference).to.equal(null);
284:        const extRef = (actions[1] as Insert).node as Element;
318:        const actions = subscribe([extRef3, extRef4]);
319:        expect(actions.length).to.equal(3);
320:        expect(actions[0]).to.satisfies(isInsert);
321:        expect(actions[1]).to.satisfies(isInsert);
322:        expect(actions[2]).to.satisfies(isInsert);
334:        const actions = subscribe({ sink, source: { fcda, controlBlock } });
335:        expect(actions.length).to.equal(1);
377:        const actions = subscribe([extRef1, extRef2, extRef3, extRef4]);
378:        expect(actions.length).to.equal(5);
379:        expect(actions[0]).to.satisfies(isInsert);
380:        expect(actions[1]).to.satisfies(isUpdate);
381:        expect((actions[1] as Update).attributes).to.deep.equal({
396:        expect(actions[2]).to.satisfies(isUpdate);
397:        expect(actions[3]).to.satisfies(isInsert);
398:        expect(actions[4]).to.satisfies(isInsert);

tExtRef/getControlBlock.ts
54: * @returns An array of update and/or remove action representing changes required

tExtRef/subscribe.ts
163:        const action = createSubscribeAction(option!, inputs);
164:        if (action)
171:        return action;
186:    .filter((action) => action) as (Insert | Update)[];
242: * @returns An array of actions to do a valid subscription

tExtRef/unsubscribe.spec.ts
24:  it("returns a remove action for each non later binding ExtRef", () => {
26:    const actions = unsubscribe(extRefs);
28:    expect(actions.length).to.equal(4);
29:    expect(actions[0]).to.satisfies(isRemove);
30:    expect((actions[0] as Remove).node).to.equal(extRefs[0]);
31:    expect(actions[1]).to.satisfies(isRemove);
32:    expect((actions[1] as Remove).node).to.equal(extRefs[1]);
33:    expect(actions[2]).to.satisfies(isRemove);
34:    expect((actions[2] as Remove).node).to.equal(extRefs[2]);
37:  it("returns an additional remove action for leave parent Input", () => {
39:    const actions = unsubscribe(extRefs);
41:    expect(actions[3]).to.satisfies(isRemove);
42:    expect((actions[3] as Remove).node).to.equal(extRefs[0].parentElement);
45:  it("returns update action for later binding type ExtRef", () => {
47:    const actions = unsubscribe(extRefs);
49:    expect(actions.length).to.equal(3);
50:    expect(actions[0]).to.satisfies(isUpdate);
51:    expect((actions[0] as Update).element).to.equal(extRefs[0]);
52:    expect(actions[1]).to.satisfies(isUpdate);
53:    expect((actions[1] as Update).element).to.equal(extRefs[1]);
54:    expect(actions[2]).to.satisfies(isUpdate);
55:    expect((actions[2] as Update).element).to.equal(extRefs[2]);
60:    const actions = unsubscribe([extRef]);
62:    expect(actions.length).to.equal(1);
63:    expect(actions[0]).to.satisfies(isUpdate);
65:    const update = actions[0] as Update;
75:    const actions = unsubscribe([extRef]);
77:    expect(actions.length).to.equal(1);
78:    expect(actions[0]).to.satisfies(isUpdate);
80:    const update = actions[0] as Update;
94:    const actions = unsubscribe(extRefs);
102:    expect(actions.length).to.equal(5);
103:    expect(actions[0]).to.satisfies(isRemove);
104:    expect((actions[0] as Remove).node).to.equal(extRefs[0]);
105:    expect(actions[1]).to.satisfies(isRemove);
106:    expect((actions[1] as Remove).node).to.equal(extRefs[1]);
107:    expect(actions[2]).to.satisfies(isUpdate);
108:    expect((actions[2] as Update).element).to.equal(extRefs[2]);
109:    expect(actions[3]).to.satisfies(isRemove);
110:    expect((actions[3] as Remove).node).to.equal(doi);
111:    expect(actions[4]).to.satisfies(isRemove);
112:    expect((actions[4] as Remove).node).to.equal(ln);
120:    const actions = unsubscribe(extRefs, { ignoreSupervision: true });
122:    expect(actions.length).to.equal(3);
123:    expect(actions[0]).to.satisfies(isRemove);
124:    expect((actions[0] as Remove).node).to.equal(extRefs[0]);
125:    expect(actions[1]).to.satisfies(isRemove);
126:    expect((actions[1] as Remove).node).to.equal(extRefs[1]);
127:    expect(actions[2]).to.satisfies(isUpdate);
128:    expect((actions[2] as Update).element).to.equal(extRefs[2]);
136:    const actions = unsubscribe([extRef]);
138:    expect(actions.length).to.equal(1);
139:    expect(actions[0]).to.satisfies(isRemove);
140:    expect((actions[0] as Remove).node).to.equal(extRef);
148:    const actions = unsubscribe([extRef]);
150:    expect(actions.length).to.equal(1);
151:    expect(actions[0]).to.satisfies(isRemove);
152:    expect((actions[0] as Remove).node).to.equal(extRef);
163:    const actions = unsubscribe(extRefs);
165:    expect(actions.length).to.equal(4);
166:    expect(actions[0]).to.satisfies(isRemove);
167:    expect((actions[0] as Remove).node).to.equal(extRefs[0]);
168:    expect(actions[1]).to.satisfies(isRemove);
169:    expect((actions[1] as Remove).node).to.equal(extRefs[1]);
170:    expect(actions[2]).to.satisfies(isRemove);
171:    expect((actions[2] as Remove).node).to.equal(extRefs[1].parentElement);
172:    expect(actions[3]).to.satisfies(isRemove);
173:    expect((actions[3] as Remove).node).to.equal(doi);
181:    const actions = unsubscribe(extRefs, { ignoreSupervision: true });
183:    expect(actions.length).to.equal(3);
184:    expect(actions[0]).to.satisfies(isRemove);
185:    expect((actions[0] as Remove).node).to.equal(extRefs[0]);
186:    expect(actions[1]).to.satisfies(isRemove);
187:    expect((actions[1] as Remove).node).to.equal(extRefs[1]);
188:    expect(actions[2]).to.satisfies(isRemove);
189:    expect((actions[2] as Remove).node).to.equal(extRefs[1].parentElement);
197:    const actions = unsubscribe(extRefs, { ignoreSupervision: true });
199:    expect(actions.length).to.equal(1);
200:    expect((actions[0] as Update).element).to.equal(extRefs[0]);
201:    expect((actions[0] as Update).attributes.serviceType).to.be.null;
209:    const actions = unsubscribe([extRefs], { ignoreSupervision: true });
211:    expect(actions.length).to.equal(1);
212:    expect((actions[0] as Update).element).to.equal(extRefs);
213:    expect((actions[0] as Update).attributes).to.not.have.own.property(

tSubstation/removeProcessElement.ts
48:/** Updates remove action for process type element
52: * @param remove - remove action on precess level elements
53: * @return - enriched remove action array

tSubstation/updateSubstation.ts
60: * @param update - update action on `Substation` attributes
61: * @returns Completed update action array */

tSubstation/updateSubstation.spec.ts
21:    const actions = updateSubstation({
26:    expect(actions.length).to.equal(1);
30:    const actions = updateSubstation({
35:    expect(actions.length).to.equal(1);
40:      const actions = updateSubstation({
45:      expect(actions.length).to.equal(1);
46:      expect(actions[0].element).to.equal(aa1);
47:      expect(actions[0].attributes).to.deep.equal({
55:      const actions = updateSubstation({
60:      expect(actions.length).to.equal(1);
64:      const actions = updateSubstation({
69:      expect(actions.length).to.equal(2);
70:      expect(actions[1].attributes).to.deep.equal({
76:      const actions = updateSubstation({
81:      expect(actions.length).to.equal(23);
82:      expect((actions[0] as Update).element).to.equal(aa1);
84:      expect((actions[1] as Update).attributes).to.deep.equal({
88:      expect((actions[2] as Update).attributes).to.deep.equal({
92:      expect((actions[3] as Update).attributes).to.deep.equal({
96:      expect((actions[4] as Update).attributes).to.deep.equal({
100:      expect((actions[5] as Update).attributes).to.deep.equal({
106:      const actions = updateSubstation({
111:      expect(actions.length).to.equal(23);
112:      expect((actions[0] as Update).element).to.equal(aa1);
114:      expect((actions[9] as Update).attributes).to.deep.equal({
119:      expect((actions[14] as Update).attributes).to.deep.equal({
124:      expect((actions[17] as Update).attributes).to.deep.equal({

tReportControl/createReportControl.spec.ts
192:      it("return valid insert action with valid datSet reference", () => {

tReportControl/updateReportControl.ts
7: * @param update - update action on `ReportControl` attributes
8: * @returns Completed update action array */

tIED/insertIED.spec.ts
9:  actions: Insert[],
13:  return actions.find(
14:    (action) =>
15:      (type === "parent" && action.parent === node) ||
16:      (type === "node" && action.node === node)
21:  actions: Insert[],
25:  return actions.filter(
26:    (action) =>
27:      (type === "parent" && (action.parent as Element).tagName === tag) ||
28:      (type === "node" && (action.node as Element).tagName === tag)

tReportControl/updateReportControl.spec.ts
20:    const actions = updateReportControl({
25:    expect(actions).to.be.empty;
31:      const actions = updateReportControl({
36:      expect(actions.length).to.equal(1);
37:      expect(actions[0]).to.satisfy(isUpdate);
38:      expect((actions[0] as Update).element).to.equal(reportControl);
39:      expect((actions[0] as Update).attributes).to.deep.equal({
47:      const actions = updateReportControl({
52:      expect(actions.length).to.equal(1);
53:      expect(actions[0]).to.satisfy(isUpdate);
54:      expect((actions[0] as Update).element).to.equal(reportControl);
55:      expect((actions[0] as Update).attributes).to.deep.equal({
65:      const actions = updateReportControl({
73:      expect(actions.length).to.equal(3);
74:      expect(actions[0]).to.satisfy(isUpdate);
75:      expect((actions[0] as Update).element).to.equal(reportControl);
76:      expect((actions[0] as Update).attributes).to.deep.equal({
82:      expect(actions[1]).to.satisfy(isUpdate);
83:      expect((actions[1] as Update).element.tagName).to.equal("ExtRef");
84:      expect((actions[1] as Update).attributes).to.deep.equal({
88:      expect(actions[2]).to.satisfy(isUpdate);
89:      expect((actions[2] as Update).element.tagName).to.equal("ExtRef");
90:      expect((actions[2] as Update).attributes).to.deep.equal({

tIED/insertIED.ts
10:  const actions: Insert[] = [];
19:    actions.push({
39:      actions.push({
53:        actions.push({
62:  return actions;
242:  const actions: (Insert | undefined)[] = [];
249:    actions.push({
259:      actions.push(addLNodeType(newIed, lNodeType, dataTypeTemplates!))
265:      actions.push(addDOType(newIed, doType, dataTypeTemplates!))
271:      actions.push(addDAType(newIed, daType, dataTypeTemplates!))
277:      actions.push(addEnumType(newIed, enumType, dataTypeTemplates!))
280:  return actions.filter((item) => item !== undefined) as Insert[];
301: * @returns An array containing diff objects representing an import IED action

tBay/updateBay.ts
64: * @param update - update action on `Bay` attributes
65: * @returns Completed update action array */

tVoltageLevel/updateVoltageLevel.ts
63: * @param update - update action on `VoltageLevel` attributes
64: * @returns Completed update action array */

tVoltageLevel/updateVoltageLevel.spec.ts
25:    const actions = updateVoltageLevel({
30:    expect(actions.length).to.equal(1);
34:    const actions = updateVoltageLevel({
39:    expect(actions.length).to.equal(1);
44:      const actions = updateVoltageLevel({
49:      expect(actions.length).to.equal(1);
50:      expect(actions[0].element).to.equal(aa1e1);
51:      expect(actions[0].attributes).to.deep.equal({
61:      const actions = updateVoltageLevel({
66:      expect(actions.length).to.equal(1);
70:      const actions = updateVoltageLevel({
75:      expect(actions.length).to.equal(2);
76:      expect(actions[1].attributes).to.deep.equal({
82:      const actions = updateVoltageLevel({
87:      expect(actions.length).to.equal(15);
88:      expect((actions[0] as Update).element).to.equal(aa1e1);
90:      expect((actions[1] as Update).attributes).to.deep.equal({
94:      expect((actions[2] as Update).attributes).to.deep.equal({
98:      expect((actions[3] as Update).attributes).to.deep.equal({
102:      expect((actions[4] as Update).attributes).to.deep.equal({
106:      expect((actions[5] as Update).attributes).to.deep.equal({
112:      const actions = updateVoltageLevel({
117:      expect(actions.length).to.equal(15);
118:      expect((actions[0] as Update).element).to.equal(aa1e1);
120:      expect((actions[7] as Update).attributes).to.deep.equal({
125:      expect((actions[8] as Update).attributes).to.deep.equal({
130:      expect((actions[12] as Update).attributes).to.deep.equal({

tBay/updateBay.spec.ts
28:    const actions = updateBay({ element: aa1, attributes: { name: "Q04" } });
30:    expect(actions.length).to.equal(1);
34:    const actions = updateBay({
39:    expect(actions.length).to.equal(1);
44:      const actions = updateBay({
49:      expect(actions.length).to.equal(1);
50:      expect(actions[0].element).to.equal(aa1e1q01);
51:      expect(actions[0].attributes).to.deep.equal({
59:      const actions = updateBay({
64:      expect(actions.length).to.equal(1);
68:      const actions = updateBay({
73:      expect(actions.length).to.equal(2);
74:      expect(actions[1].attributes).to.deep.equal({
80:      const actions = updateBay({
85:      expect(actions.length).to.equal(6);
86:      expect(actions[0]).to.satisfy(isUpdate);
87:      expect((actions[0] as Update).element).to.equal(aa1e1q01);
89:      expect(actions[1]).to.satisfy(isUpdate);
90:      expect((actions[1] as Update).attributes).to.deep.equal({
94:      expect(actions[2]).to.satisfy(isUpdate);
95:      expect((actions[2] as Update).attributes).to.deep.equal({
101:      const actions = updateBay({
106:      expect(actions.length).to.equal(6);
107:      expect(actions[0]).to.satisfy(isUpdate);
108:      expect((actions[0] as Update).element).to.equal(aa1e1q01);
110:      expect(actions[3]).to.satisfy(isUpdate);
111:      expect((actions[3] as Update).attributes).to.deep.equal({
116:      expect(actions[4]).to.satisfy(isUpdate);
117:      expect((actions[4] as Update).attributes).to.deep.equal({
122:      expect(actions[5]).to.satisfy(isUpdate);
123:      expect((actions[5] as Update).attributes).to.deep.equal({
130:      const actions = updateBay({
135:      expect(actions.length).to.equal(5);
```

### PR DESCRIPTION
**Pull request by danyill**
Closes #50

### CODE REVIEW

**Review by JakobVogelsang — COMMENTED** [review]
Aside one change I would like to request the rest is comments and could go in as is. Thank you!!

**Comment by JakobVogelsang** [comment]
Closes #50

File: tExtRef/subscribe.ts [hunk 1]
```diff
@@ -145,47 +145,45 @@ function createSubscribeAction(
...
 
-function createSubscribeActions(
-  connections: Connection[]
-): (Insert | Update)[] {
-  const inputActions: Insert[] = [];
+function createSubscribeedits(connections: Connection[]): (Insert | Update)[] {
+  const inputedits: Insert[] = [];
 
-  const extRefActions = connections
+  const extRefedits = connections
```
[the remark this answers is not in the document]
Comment (danyill) [reply tExtRef/subscribe.ts]: I just didn't do a very good check on some of my find and replace and could have done it more cleverly, sorry... :shrug:

**Comment by danyill** [comment]
Should be better now, thanks for the review :+1:

**Review by JakobVogelsang — APPROVED** [review]
LGTM

**Pull request merged**